lib/src/revset.rs

```diff
@@ -2669,11 +2688,23 @@ impl<'a> RevsetParseContext<'a> {
         extensions: &'a RevsetExtensions,
         workspace: Option<RevsetWorkspaceContext<'a>>,
     ) -> Self {
+        let mut function_map = HashMap::<&'static str, &'a RevsetFunction>::new();
+        for (name, func) in BUILTIN_FUNCTION_MAP.iter() {
+            function_map.insert(name, func);
+        }
+        for (name, func) in &extensions.custom_functions {
+            match function_map.entry(name) {
+                hash_map::Entry::Occupied(_) => panic!("Cannot override builtin function '{name}'"),
+                hash_map::Entry::Vacant(v) => v.insert(func),
+            };
```

Comment:
nit: maybe better to detect name conflicts at `add_custom_extension()` so that bad extension can be blamed based on call stack.